shell/src/main/java/alluxio/cli/fsadmin/pathconf/ShowCommand.java

```diff
@@ -0,0 +1,143 @@
+import alluxio.wire.Property;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Show path level configurations.
+ *
+ * <p>
+ * It has two modes:
+ * 1. only show configurations for the specific path;
+ * 2. show all configurations applicable to the path.
+ * <p>
+ * For example, suppose there are two path level configurations:
+ * /a: property1=value1
+ * /a/b: property2=value2
+ * Then in mode 1, only property2=value2 will be shown for /a/b,
+ * but in mode 2, since /a is the prefix of /a/b, both properties will be shown for /a/b.
+ */
+public final class ShowCommand extends AbstractFsAdminCommand {
+  public static final String RESOLVE_OPTION_NAME = "resolve";
+
+  private static final Option RESOLVE_OPTION =
+      Option.builder()
+          .longOpt(RESOLVE_OPTION_NAME)
+          .required(false)
+          .hasArg(false)
+          .desc("Whether to show all path level configurations applicable to the path")
+          .build();
+
+  /**
+   * @param context     fsadmin command context
+   * @param alluxioConf Alluxio configuration
+   */
+  public ShowCommand(Context context, AlluxioConfiguration alluxioConf) {
+    super(context);
+  }
+
+  @Override
+  public String getCommandName() {
+    return "show";
+  }
+
+  @Override
+  public Options getOptions() {
+    return new Options()
+        .addOption(RESOLVE_OPTION);
+  }
+
+  @Override
+  public void validateArgs(CommandLine cl) throws InvalidArgumentException {
+    CommandUtils.checkNumOfArgsNoLessThan(this, cl, 1);
+  }
+
+  private String format(String key, String value) {
+    return key + " = " + value;
+  }
+
+  @Override
+  public int run(CommandLine cl) throws IOException {
+    String targetPath = cl.getArgs()[1];
+    Configuration configuration = mMetaConfigClient.getConfiguration();
+
+    if (cl.hasOption(RESOLVE_OPTION_NAME)) {
+      Set<PropertyKey> pathPropertyKeys = new HashSet<>();
+      Map<String, AlluxioConfiguration> pathConfMap = new HashMap<>();
+
+      configuration.getPathConf().forEach((path, conf) -> {
+        AlluxioProperties properties = new AlluxioProperties();
+        conf.forEach(property -> {
+          pathPropertyKeys.add(PropertyKey.fromString(property.getName()));
+          properties.set(PropertyKey.fromString(property.getName()),
+              property.getValue());
+        });
+        pathConfMap.put(path, new InstancedConfiguration(properties));
+      });
+      PathConfiguration pathConf = PathConfiguration.create(pathConfMap);
+
+      AlluxioURI targetUri = new AlluxioURI(targetPath);
+      List<PropertyKey> propertyKeys = new ArrayList<>(pathPropertyKeys);
+      propertyKeys.sort(Comparator.comparing(PropertyKey::getName));
```

Comment:
could we build a list of just the relevant properties, then sort those by name? Then we don't need `propertyKeys` or `pathPropertyKeys` at all

Reply:
The problem is that all properties might be "relevant", for example, if there is no path level configuration set for path `/a/b/c/d`, but there are configurations for path `/a`, `/a/b`, `/a/b/c`, then all properties in these configurations are "relevant".

Reply:
What do you think of adding a `List<PropertyKey> getAllProperties(uri)` method in `PathConfiguration`? I could see this being useful for debugging or other code in the future.